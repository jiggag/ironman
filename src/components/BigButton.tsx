import React, { useMemo } from 'react';
import {
  Text, TouchableOpacity, StyleSheet, StyleProp, ViewStyle, TextStyle,
} from 'react-native';
import Constant from '@utils/constants';

interface ButtonProps {
  onPress: () => void;
  text: string;
  buttonStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
}
export const BigButton: React.FC<ButtonProps> = ({
  onPress, text, buttonStyle, textStyle,
}) => {
  const customButtonStyle = useMemo<StyleProp<ViewStyle>>(() => [styles.button, buttonStyle], [buttonStyle]);
  return (
    <TouchableOpacity onPress={onPress} activeOpacity={0.5} style={customButtonStyle}>
      <Text style={[styles.buttonText, textStyle]}>{text}</Text>
    </TouchableOpacity>
  );
};

export const MainBigButton: React.FC<ButtonProps> = ({ text, onPress }) => {
  return <BigButton text={text} onPress={onPress} buttonStyle={styles.mainButton} textStyle={styles.mainButtonText} />;
};

export const TextLineButton: React.FC<ButtonProps> = ({ text, onPress }) => {
  return (
    <BigButton
      text={text}
      onPress={onPress}
      buttonStyle={styles.textLineButton}
      textStyle={styles.textLineButtonText}
    />
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: Constant.MAIN_COLOR,
    borderRadius: 2,
    elevation: 4,
    marginHorizontal: 20,
    marginVertical: 8,
    paddingVertical: 14,
    shadowColor: Constant.SHADOW_COLOR,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.5,
    shadowRadius: 8,
  },
  buttonText: {
    color: Constant.WHITE_COLOR,
    fontSize: 16,
    fontWeight: '600',
  },
  mainButton: {
    backgroundColor: Constant.MAIN_COLOR,
    borderRadius: 2,
    borderWidth: 0,
    paddingVertical: 20,
  },
  mainButtonText: {
    color: Constant.WHITE_COLOR,
    fontWeight: 'bold',
  },
  textLineButton: {
    backgroundColor: Constant.WHITE_COLOR,
    elevation: 0,
    paddingVertical: 20,
    shadowOpacity: 0,
  },
  textLineButtonText: {
    color: Constant.MAIN_COLOR,
    fontWeight: 'bold',
    textDecorationLine: 'underline',
  },
});
