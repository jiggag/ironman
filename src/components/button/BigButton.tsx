import React, { useMemo } from 'react';
import {
  Text, TouchableOpacity, StyleSheet, StyleProp, ViewStyle, TextStyle,
} from 'react-native';
import Constant, { FontSize, FontWeight } from '@utils/constants';

export interface ButtonProps {
  onPress: () => void;
  text: string;
  buttonStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
}

export const BigButton: React.FC<ButtonProps> = ({
  onPress, text, buttonStyle, textStyle,
}) => {
  const customButtonStyle = useMemo<StyleProp<ViewStyle>>(() => [styles.button, buttonStyle], [buttonStyle]);

  return (
    <TouchableOpacity onPress={onPress} activeOpacity={0.5} style={customButtonStyle}>
      <Text style={[styles.buttonText, textStyle]}>{text}</Text>
    </TouchableOpacity>
  );
};

export const MainBigButton: React.FC<ButtonProps> = ({ text, onPress }) => {
  return <BigButton text={text} onPress={onPress} buttonStyle={styles.mainButton} textStyle={styles.mainButtonText} />;
};

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: Constant.MAIN_COLOR,
    borderRadius: 2,
    elevation: 4,
    marginHorizontal: 20,
    marginVertical: 8,
    paddingVertical: 14,
    shadowColor: Constant.SHADOW_COLOR,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.5,
    shadowRadius: 8,
  },
  buttonText: {
    color: Constant.WHITE_COLOR,
    fontSize: FontSize.button,
    fontWeight: FontWeight.bold,
  },
  mainButton: {
    backgroundColor: Constant.MAIN_COLOR,
    borderRadius: 2,
    borderWidth: 0,
    paddingVertical: 20,
  },
  mainButtonText: {
    color: Constant.WHITE_COLOR,
    fontWeight: FontWeight.bold,
  },
});
