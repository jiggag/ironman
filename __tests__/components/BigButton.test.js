import React from 'react';
import { render } from '@testing-library/react-native';
import BigButton from '../../src/components/BigButton';

it('Renderer Component: BigButton', () => {
  const { baseElement } = render(
    <BigButton onPress={() => {}} text="버튼" buttonStyle={{
      backgroundColor: '#ffffff'
    }} />
  )
  expect(baseElement).toMatchSnapshot();
});