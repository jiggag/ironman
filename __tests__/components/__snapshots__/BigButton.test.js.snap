// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Component: BigButton 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "elevation": 2,
      "marginHorizontal": 20,
      "marginVertical": 8,
      "opacity": 1,
      "paddingVertical": 14,
      "shadowColor": "#333333",
      "shadowOffset": Object {
        "height": 4,
        "width": 6,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    버튼
  </Text>
</View>
`;
